packages/strapi/lib/middlewares/index.js

```diff
@@ -3,9 +3,19 @@
 const glob = require('glob');
 const path = require('path');
 const { parallel } = require('async');
-const { after, includes, indexOf, drop, dropRight, uniq, defaultsDeep, get, set, isEmpty, isUndefined, union } = require('lodash');
+const { after, includes, indexOf, drop, dropRight, uniq, defaultsDeep, get, set, isEmpty, isUndefined, union, keys } = require('lodash');
 
 module.exports = function() {
+  // Set if is admin destination for middleware application.
+  // TODO: Use dynamic config for admin url.
+  strapi.app.use(async (ctx, next) => {
+    const accepted = keys(strapi.plugins).map(url => `^\/${url}`);
```

Comment:
I think you could make this work out of the Koa stack. It will always return the same result. Also, you don't need to use the Lodash `keys` method. The `Object.keys` provides great performances since the latest Node.js release. Same things for the usage of the `set` method few lines below.

It means that we could reduce the middleware to this:

```js
const accepted = keys(strapi.plugins).map(url => `^\/${url}`).concat(['^\/admin']);

strapi.app.use(async (ctx, next) => {
  ctx.request.admin = accepted.some(rx => new RegExp(rx).test(ctx.request.url));

  return next();
});
```